### ISSUE TITLE
MacWrite 2.2 causes the emulator to crash with "index out of bounds" error

### ISSUE THREAD

**Issue by ssg** [mihaip/infinite-mac#127]
Trying to run MacWrite 2.2 crashes the emulator on any version of System.

**Comment by ssg** [comment]
You're right, it worked on 6.0 for me as well. But it fails on 7.1 (https://infinitemac.org/1992/System%207.1) for example. The browser is Firefox (111.0.1 64-bit on Windows).

![image](https://user-images.githubusercontent.com/241217/228674226-425ee3d2-96b8-4f96-98f8-97ef29516ab8.png)

**Comment by that-ben** [comment]
OK, well that's a step forward in understanding the issue, but you forgot to answer my questions 2 and 3.

**Comment by ssg** [comment]
You're right. It works when I select Mac II, Mac Plus, or Mac SE on the same emulator, but it keeps crashing with Mac IIfx. Is that expected?

**Comment by ssg** [comment]
IIfx also seems to have 68030 so it might also be causing the compatibility problem as you said.

That's okay of course, but the emulator crashing with "index out of bounds" error isn't very informative. I thought this was a bug in the emulator code because it didn't even know what actually went wrong. So, it's still different than how Basilisk II handles the situation (Application quit message).

I don't think the emulator should crash completely because of app incompatibility.

**Comment by that-ben** [comment]
The Mac IIfx has a very fast 40mhz 68030 CPU, so it might be normal.  I'm not 100% sure, but I would totally believe it's normal.  Also, it's always best to use the apps on an OS that fits the app's era.  In the case of MacWrite 2.2, it was released in 1984 so IMO, it's not appropriate to run it on a decade newer OS such as System 7.1.  It really belongs to the 1980's OS'es such as System 6 or preferably even older than that.

If you'd like to use MacWrite under 68030 or 68040 architecture and System 7, Mac OS 8 or even Mac OS 9, have you considered trying Claris MacWrite? (also known as MacWrite 5.0)

**Comment by ssg** [comment]
Thank you for the suggestions, but I'm running apps out of mere curiosity and nostaliga. I don't intend on doing any serious work on them. I just wanted to report a bug that I noticed if it could be considered that. :) 

And sorry for not giving enough information earlier. I really thought it happened consistently on all emulator flavors, but I wasn't investigating it seriously so I turned out wrong. I'll be more careful the next time.

**Comment by that-ben** [comment]
@mihaip  TL;DR instead of halting the emulator on an application crash, ssg would like Mac OS to display its crash window with error type 25 or whatever it is.  (see screenshot 3 messages above this one)
